R/derive_var_joined_exist_flag.R

```diff
@@ -406,6 +407,58 @@
 #'   first_cond_upper = val.join == "++",
 #'   filter_join = val == "0" & all(val.join %in% c("+", "++"))
 #' )
+#'
+#' # flag each dose which is lower than the previous dose per subject
+#' ex <- tribble(
+#'   ~USUBJID, ~EXSTDTM,          ~EXDOSE,
+#'   "1",      "2024-01-01T08:00",      2,
+#'   "1",      "2024-01-02T08:00",      4,
+#'   "2",      "2024-01-01T08:30",      1,
+#'   "2",      "2024-01-02T08:30",      4,
+#'   "2",      "2024-01-03T08:30",      3,
+#'   "2",      "2024-01-04T08:30",      2
```

Comment:
I would add a record to show that only the adjacent dose is considered.
```suggestion
#'   "2",      "2024-01-04T08:30",      2,
#'   "2",      "2024-01-05T08:30",      2,
```